### DISCUSSION TITLE
Repo Relative Path Pt2

### DISCUSSION THREAD

**Discussion by GavinOsborn** [JanDeDobbeleer/oh-my-posh#2767] [Troubleshoot]
Hey, me again!
In #2717 we introduced the `.Git.Dir` variable that made it possible to build repo relative paths - thank you 😊.
I'm wondering how feasible it would be to dial that up to eleven and support formatting  the repo relative path with the same formatting options and styles generally supported by the `path` segment?

Lets say I have a git repo located at: `~/code/my-repo` and my terminal session is currently navigating `~/code/my-repo/source/website/images`. Thanks to your previous update I can now calculate a repo relative path `/source/website/images`. What I don't seem to be able to do is then apply styles (`agnoster`, etc) to that path - I can only ever get the raw value.

I can think of a couple of mental models for doing it but I've been unable to figure out a way to do it in v9.
1) You could either configure your `path` segment with the location we want it to consider the root of the path being evaluated (`.Git.Dir`)
2) the `path` segment could be updated to format non-rooted values

In either case I haven't figured out how to do this myself yet.

Is this already possible?
If not - does this seem like something that could realistically be supported?

**Comment by JanDeDobbeleer** [comment]
> I'm wondering how feasible it would be to dial that up to eleven and support formatting the repo relative path with the same formatting options and styles generally supported by the path segment?

Not that trivial tbh. The main issue is that for this to work we need template functions to have access to the environment, and that's not easily achievable (in a clean, testable way).

But, to be clear, this only needs to work in the path segment, right? So potentially, adding a function in the path segment that allows formatting the relative `.Git.Dir` would be sufficient?
